src/modules/job-info/watch.c

```diff
@@ -183,12 +283,22 @@ void watch_cb (flux_t *h, flux_msg_handler_t *mh,
         errmsg = "eventlog-watch request rejected without streaming RPC flag";
         goto error;
     }
-    if (!(w = watch_ctx_create (ctx, msg, id)))
-        goto error;
 
-    if (watch_key (w) < 0)
+    (void)flux_request_unpack (msg, NULL, "{s:b}", "guest", &guest);
+    (void)flux_request_unpack (msg, NULL, "{s:s}", "path", &path);
+
+    if (!(w = watch_ctx_create (ctx, msg, id, guest, path)))
         goto error;
 
+    if (path && strcasecmp (path, "eventlog")) {
```

Comment:
Whenever I see `strcmp` or `strcasecmp` without an explicit `== 0` or `!= 0` I have little alarm bells. In this case it looks like you want `== 0` but I admit I don't fully understand the code. Either way would you  be willing to either make the test explicit, or at least add a comment about what is being checked?